### PR TITLE
Merge v0.6.6

### PR DESCRIPTION
**Pull request by iproudhon**
Merge v0.6.6

### CODE REVIEW

File: db.go [hunk 1]
```diff
@@ -37,15 +33,13 @@ const (
 	//   - requires gcc
 	//   - use rocksdb build tag (go build -tags rocksdb)
 	RocksDBBackend BackendType = "rocksdb"
-	// BadgerDBBackend represents badger (uses github.com/dgraph-io/badger/v2)
-	//   - EXPERIMENTAL
-	//   - use badgerdb build tag (go build -tags badgerdb)
+
 	BadgerDBBackend BackendType = "badgerdb"
 	// Alternative rocksdb
```
Comment (zemyblue) [inline db.go:38]: lint error
`RDBBackend represents alternative rocksdb.`

File: go.mod [hunk 2]
```diff
@@ -10,6 +10,10 @@ require (
...
 	github.com/syndtr/goleveldb v1.0.1-0.20200815110645-5c35d600f0ca
-	go.etcd.io/bbolt v1.3.5
-	google.golang.org/grpc v1.35.0
+	go.etcd.io/bbolt v1.3.6
+	google.golang.org/grpc v1.42.0
 )
+
+// FIXME: gorocksdb bindings for OptimisticTransactionDB are not merged upstream, so we use a fork
+// See https://github.com/tecbot/gorocksdb/pull/216
+// replace github.com/tecbot/gorocksdb => github.com/roysc/gorocksdb v1.1.1
```
Comment (zemyblue) [inline go.mod:19]: Why is this line commented out?

File: prefixdb_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	require.NoError(t, db.Set(bz("key2"), bz("value2")))
+	require.NoError(t, db.Set(bz("key3"), bz("value3")))
+	require.NoError(t, db.Set(bz("something"), bz("else")))
+	require.NoError(t, db.Set(bz("k"), bz("val")))
+	require.NoError(t, db.Set(bz("ke"), bz("valu")))
+	require.NoError(t, db.Set(bz("kee"), bz("valuu")))
+	return db
+}
+
+func checkDomain(t *testing.T, itr Iterator, start, end []byte) {
```
Comment (zemyblue) [inline prefixdb_test.go:23]: `checkDomain` is empty function.
WHAT IS `checkDoman'` the rule

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,30 @@
...
+## 0.6.5
+
+**2021-08-04**
+
+**Important note**: This version was tagged by accident, and should not be
+used. The tag now points to the [package-reorg
+branch](https://github.com/tendermint/tm-db/tree/package-reorg) so that
+any existing dependencies should not break.
+
+## 0.6.4
```
Comment (zemyblue) [inline CHANGELOG.md:27]: Please add only our changes and link to the original change log with a brief description.

File: prefixdb_test.go (hunk 3 above)
Comment (iproudhon) [reply prefixdb_test.go:23]: We removed checkDomain(). However, prefixdb_test.db uses that function. I added empty function to work around.

File: go.mod (hunk 2 above)
Comment (iproudhon) [reply go.mod:19]: Because we use line/gorocksdb, not tecbot/gorocksdb or roysc/gorocksdb.

File: CHANGELOG.md (hunk 4 above)
Comment (iproudhon) [reply CHANGELOG.md]: updated

**Comment by zemyblue** [comment]
@iproudhon , please use "Merge pull request" when merge.

**Review by zemyblue — APPROVED** [review]
Thank you for your effort.

File: rdb_test.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+	_, ok := db.(*RDB)
+	assert.True(t, ok)
+}
+
+func TestRDBStats(t *testing.T) {
+	name := fmt.Sprintf("test_%x", randStr(12))
+	dir := os.TempDir()
+	db, err := NewDB(name, RDBBackend, dir)
+	defer cleanupDBDir(dir, name)
+	require.NoError(t, err)
```
Comment (0Tech) [inline rdb_test.go:31]: Is `defer` in the correct order? I saw almost same codes, but `defer` follows `require.NoError()`.
If the reader (human) does not know the internals of `NewDB()` and `cleanupDBDir()`, using `os.RemoveAll()` right after `os.TempDir()` could be a better choice. I saw it in many places in this PR.
If you accept my opinion, I also ask you to fix all the similar lines in this PR.

**Review by dudong2 — APPROVED** [review]

**Pull request merged**